app/Http/Controllers/CourseController.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Course;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\Auth;
+
+class CourseController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\View\View
+     */
+    public function index(): View
+    {
+        return view('courses.index', [
+            'courses' => Course::all(),
+        ]);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  \App\Models\Course  $course
+     * @return \Illuminate\View\View
+     */
+    public function show(Course $course)
+    {
+        $user = Auth::user();
+        $courseUser = $user->courses->where('id', $course->id)->first();
```

Comment:
This currently breaks if a user isn't logged in. I think the issue here is more that we shouldn't allow guests to access courses. We'll need middleware on these routes accordingly.